saw-core/src/SAWCore/Prim.hs

```diff
@@ -168,8 +164,10 @@ bvCountTrailingZeros _ (BV w x) = BV w (toInteger (go 0))
 
 -- | @at@ specialized to BitVector (big-endian)
 -- at :: (n :: Nat) -> (a :: sort 0) -> Vec n a -> Nat -> a;
-at_bv :: Int -> () -> BitVector -> Natural -> Bool
-at_bv _ _ x i = testBit (unsigned x) (width x - 1 - fromIntegral i)
+at_bv :: () -> () -> BitVector -> Int -> Maybe Bool
+at_bv _ _ x i
+  | i < width x = Just (testBit (unsigned x) (width x - 1 - i))
+  | otherwise = Nothing
```

Comment:
This is the key change needed for better performance, right? Would it be worth leaving a comment here stating the performance-related impact of checking `i < width x` first?

Reply:
I just rebased the branch and re-pushed with an updated commit message that thoroughly explains the changes. When this PR is merged I'll plan to close #2856, but I'll open a new tech-debt issue asking to rewrite `SAWCore.Conversion` without the impenetrable typeclass magic.

Reply:
Note added in 2f58c189880a4423c0137d2d53569d7d40990daf.